File: solutions/python/binary-search/1/binary_search.py
def find(search_list: list[int], value: int) -> int:
    left = 0
    right = len(search_list) - 1
    while left <= right:
        mid = (left + right) // 2
        if search_list[mid] == value:
            return mid
        if search_list[mid] < value:
            left = mid + 1
        else:
            right = mid - 1
    raise ValueError("value not in array")

File: solutions/c/binary-search/1/binary_search.c
#include "binary_search.h"

const int* binary_search(int value, const int* arr, size_t length) {
    if (!arr) return NULL;
    const int* left = arr;
    const int* right = arr + length - 1;
    while (left <= right) {
        const int* mid = left + (right - left) / 2;
        if (*mid == value) return mid;
        if (*mid < value)
            left = mid + 1;
        else
            right = mid - 1;
    }
    return NULL;
}
